### PR TITLE
chruby-fish: init at 0.8.2

### PR DESCRIPTION
**Pull request by cohei**
###### Motivation for this change

Add chruby-fish to nixpkgs.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] ~~Tested execution of all binary files (usually in `./result/bin/`)~~ no binary files
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/misc/chruby-fish/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    repo = "chruby-fish";
+    rev = "v${version}";
+    sha256 = "15q0ywsn9pcypbpvlq0wb41x4igxm9bsvhg9a05dqw1n437qjhyb";
+  };
+
+  installFlags = [ "PREFIX=$(out)" ];
+
+  meta = {
+    description = "Thin wrapper around chruby to make it work with the Fish shell";
+    homepage = "https://github.com/JeanMertz/chruby-fish";
```
Comment (cab404) [inline pkgs/development/tools/misc/chruby-fish/default.nix:19]: Doesn't that means that you should wrap this with chruby?

Comment (cohei) [reply pkgs/development/tools/misc/chruby-fish/default.nix:19]: You mean something like installing both chruby and fish automatically enables chruby-fish? I don't know how to achieve that...

With curent implementation, we can at least use chruby-fish without wrapping by

1. `nix-env --install fish chruby chruby-fish`
1. write some config like [here](https://github.com/cohei/dotfiles/blob/cf1d6cb9a20633e4646cbc983e63283d108cedf8/home/.config/fish/config.fish#L29-L31)

**Comment by cab404** [comment]
I mean installing chruby-fish means chruby is also added

**Comment by cohei** [comment]
That is convenient but not necessary to install chruby-fish. So I attempted to keep dependencies minimum. For example, when someone is using both chruby and chruby-fish for bash and fish respectively, this may avoid trouble.

**Comment by cab404** [comment]
> That is convenient but not necessary to install chruby-fish. So I attempted to keep dependencies minimum. For example, when someone is using both chruby and chruby-fish for bash and fish respectively, this may avoid trouble.

ehm, I meant that you can add chruby to chruby-fish path. there will be no chruby in shell path.

**Comment by cohei** [comment]
Like https://github.com/NixOS/nixpkgs/commit/6039f5e44033b3727b2758d73f290d457cfa32d5 ?

**Comment by cab404** [comment]
> Like [6039f5e](https://github.com/NixOS/nixpkgs/commit/6039f5e44033b3727b2758d73f290d457cfa32d5) ?

Nope. Like this: https://github.com/NixOS/nixpkgs/blob/master/pkgs/misc/emulators/mednaffe/default.nix#L21

Add makeWrapper to your buildInputs, and run `wrapProgram $out/bin/chruby-fish --set PATH ${chruby}/bin` in postInstall phase.

**Comment by cohei** [comment]
Thank you for the example! I'll try.

**Comment by cohei** [comment]
> Add makeWrapper to your buildInputs, and run `wrapProgram $out/bin/chruby-fish --set PATH ${chruby}/bin` in postInstall phase.

`chruby-fish`'s main resources are `share/chruby/{chruby,auto}.fish`. There is no executable, so it is impossible to do `wrapPropram` like this.

Is there anything else I can do?

**Comment by cab404** [comment]
Sorry for everything you've already been through to merge the stuff

E.g can do sed -i 's\chruby\${chruby}/bin/chruby\'.

**Review by cab404 — APPROVED** [review]

**Comment by cab404** [comment]
Result of `nixpkgs-review pr 94345` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>chruby-fish</li>
  </ul>
</details>

**Comment by cab404** [comment]
I read a bit more. Turns out, chruby is a script itself, and doesn't pull whole ruby with itself)

Although it's totally fine as a package, it still requires CHRUBY_ROOT environment variable. It would be nice to set it inside auto.fish and chruby.fish on patchPhase.

**Comment by cab404** [comment]
```diff
diff --git a/pkgs/development/tools/misc/chruby-fish/default.nix b/pkgs/development/tools/misc/chruby-fish/default.nix
index b8acc475204..8e552b4a9a5 100644
--- a/pkgs/development/tools/misc/chruby-fish/default.nix
+++ b/pkgs/development/tools/misc/chruby-fish/default.nix
@@ -1,4 +1,4 @@
-{ stdenv, lib, fetchFromGitHub }:
+{ stdenv, lib, fetchFromGitHub, chruby }:

 stdenv.mkDerivation rec {
   pname = "chruby-fish";
@@ -12,6 +12,11 @@ stdenv.mkDerivation rec {
     sha256 = "15q0ywsn9pcypbpvlq0wb41x4igxm9bsvhg9a05dqw1n437qjhyb";
   };

+  postInstall = ''
+    sed -i -e '1cset CHRUBY_ROOT ${chruby}' $out/share/chruby/auto.fish
+    sed -i -e '1cset CHRUBY_ROOT ${chruby}' $out/share/chruby/chruby.fish
+  '';
+
   installFlags = [ "PREFIX=$(out)" ];

   meta = {

```

I can't edit your PR, so here's a patch)

**Comment by cab404** [comment]
@cohei ?

File: pkgs/development/tools/misc/chruby-fish/default.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  src = fetchFromGitHub {
+    owner = "JeanMertz";
+    repo = "chruby-fish";
+    rev = "v${version}";
+    sha256 = "15q0ywsn9pcypbpvlq0wb41x4igxm9bsvhg9a05dqw1n437qjhyb";
+  };
+
+  postInstall = ''
+    sed -i -e '1iset CHRUBY_ROOT ${chruby}' $out/share/chruby/auto.fish
+    sed -i -e '1iset CHRUBY_ROOT ${chruby}' $out/share/chruby/chruby.fish
```
Comment (cohei) [inline pkgs/development/tools/misc/chruby-fish/default.nix:17]: I used `i` commands instead of `c` to keep original contents.

**Comment by cohei** [comment]
Never mind! Thanks to the detour, I understand Nix more than before.

**Comment by cab404** [comment]
Also, you can post this PR [here](https://discourse.nixos.org/t/prs-already-reviewed/2617/218), and there will be contributors with merge flag

**Comment by cohei** [comment]
They say

> Select an open pull request that is not your own

Could you post it there instead, if you don't mind?

**Comment by cab404** [comment]
Np, didn't know about this)

File: pkgs/development/tools/misc/chruby-fish/default.nix [hunk 3]
```diff
@@ -0,0 +1,29 @@
+{ stdenv, lib, fetchFromGitHub, chruby }:
+
+stdenv.mkDerivation rec {
+  pname = "chruby-fish";
+
```
Comment (AndersonTorres) [inline pkgs/development/tools/misc/chruby-fish/default.nix:5]:
```suggestion
```

Comment (AndersonTorres) [reply pkgs/development/tools/misc/chruby-fish/default.nix:5]: Extra whitespace

**Pull request merged**